ocean_lib/data_provider/data_service_provider.py

```diff
@@ -633,6 +634,14 @@ def _http_method(method, *args, **kwargs):
             )
             raise
 
+    @staticmethod
+    def check_single_file_info(file_url, provider_uri=None):
+        _, endpoint = DataServiceProvider.build_fileinfo(provider_uri)
+        data = {"url": file_url}
+        response = requests.post(endpoint, json=data).json()
```

Comment:
A "check" function should be doing reads, not writes. Yet by calling HTTP POST, it's doing a write. Why not use HTTP GET?

Reply:
This was discussed on implementation.
1. We need a payload since sometimes we are requesting data on ddo's (encrypted), otherwise on raw URLs. This can not be achieved with GET as clearly as with POST.
2. This is not an idempotent request, which is a prerequisite for GET. We can not guarantee the same status of the file URL: it might be deleted, moved or otherwise inaccessible to the exterior (which is exactly what we are checking). As far as I know, POST is ok for requests that are not idempotent.

Reply:
OK! Thanks for the clear answer. When you do the squash & merge, please give the info you gave here, it will help.